modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set(pdl_dir             ${ZEPHYR_HAL_INFINEON_MODULE_DIR}/mtb-pdl-cat1)
+set(pdl_drv_dir         ${ZEPHYR_HAL_INFINEON_MODULE_DIR}/mtb-pdl-cat1/drivers)
+set(pdl_dev_cat1a_dir   ${ZEPHYR_HAL_INFINEON_MODULE_DIR}/mtb-pdl-cat1/devices/COMPONENT_CAT1A)
+
+zephyr_compile_definitions(${CONFIG_SOC_PART_NUMBER})
+
+# Add mtb-pdl-cat1
+zephyr_include_directories(${pdl_drv_dir}/include)
+zephyr_include_directories(${pdl_dev_cat1a_dir}/include)
+zephyr_include_directories(${pdl_dev_cat1a_dir}/include/ip)
+zephyr_include_directories(${pdl_dev_cat1a_dir}/templates/COMPONENT_MTB)
+
+zephyr_library_sources(${pdl_dev_cat1a_dir}/source/cy_device.c)
+zephyr_library_sources(${pdl_drv_dir}/source/TOOLCHAIN_GCC_ARM/cy_syslib_gcc.S)
+
+zephyr_library_sources_ifdef(CONFIG_CPU_CORTEX_M4
+  ${pdl_dev_cat1a_dir}/templates/COMPONENT_MTB/COMPONENT_CM4/system_psoc6_cm4.c)
+zephyr_library_sources_ifdef(CONFIG_CPU_CORTEX_M0PLUS
+  ${pdl_dev_cat1a_dir}/templates/COMPONENT_MTB/COMPONENT_CM0P/system_psoc6_cm0plus.c)
+
+# Peripheral drivers
+zephyr_library_sources_ifdef(CONFIG_SERIAL ${pdl_drv_dir}/source/cy_scb_uart.c)
+zephyr_library_sources_ifdef(CONFIG_ADC ${pdl_drv_dir}/source/cy_sar.c)
+zephyr_library_sources_ifdef(CONFIG_I2C ${pdl_drv_dir}/source/cy_scb_i2c.c)
+zephyr_library_sources_ifdef(CONFIG_SPI ${pdl_drv_dir}/source/cy_scb_spi.c)
+zephyr_library_sources_ifdef(CONFIG_COUNTER ${pdl_drv_dir}/source/cy_tcpwm_counter.c)
+zephyr_library_sources_ifdef(CONFIG_PWM ${pdl_drv_dir}/source/cy_tcpwm_pwm.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_PSOC6 ${pdl_drv_dir}/source/cy_sysint.c)
+
+if(CONFIG_SERIAL OR CONFIG_I2C OR CONFIG_SPI)
+  zephyr_library_sources(${pdl_drv_dir}/source/cy_scb_common.c)
+endif()
+
+if(CONFIG_ADC)
+  zephyr_library_sources(${pdl_drv_dir}/source/cy_sysanalog.c)
+endif()
```

Comment:
please don't use generic symbols to enable conditional compilation, see how other HALs do this using hidden selectable symbols, STM32 or GD32 HALs. Seems like this is addressed in next commit: just squash changes, there's no need to keep a wrong intermediate solution in git history.

Reply:
yes i done in https://github.com/zephyrproject-rtos/zephyr/pull/43494/commits/c75442ac81197b8a3936e09d0b73836eaab1c6ba. I will squash.

Reply:
squashed.